packages/contracts-bedrock/src/L1/StandardValidator.sol

```diff
@@ -0,0 +1,271 @@
+        IOptimismMintableERC20Factory _factory = IOptimismMintableERC20Factory(_sysCfg.optimismMintableERC20Factory());
+        internalRequire(stringEq(_factory.version(), optimismMintableERC20FactoryVersion), "MERC20F-10");
+        internalRequire(bytecodeHashEq(address(_factory), optimismMintableERC20FactoryHash), "MERC20F-20");
+
+        IL1StandardBridge _bridge = IL1StandardBridge(payable(_sysCfg.l1StandardBridge()));
+        internalRequire(_factory.BRIDGE() == address(_bridge), "MERC20F-30");
+        internalRequire(_factory.bridge() == address(_bridge), "MERC20F-40");
+    }
+
+    function assertValidL1ERC721Bridge(ISystemConfig _sysCfg) internal {
+        IL1ERC721Bridge _bridge = IL1ERC721Bridge(_sysCfg.l1ERC721Bridge());
+        internalRequire(stringEq(_bridge.version(), l1ERC721BridgeVersion), "L721B-10");
+        internalRequire(bytecodeHashEq(address(_bridge), l1ERC721BridgeHash), "L721B-20");
+
+        IL1CrossDomainMessenger _l1XDM = IL1CrossDomainMessenger(_sysCfg.l1CrossDomainMessenger());
+        internalRequire(address(_bridge.OTHER_BRIDGE()) == Predeploys.L2_ERC721_BRIDGE, "L721B-30");
+        internalRequire(address(_bridge.otherBridge()) == Predeploys.L2_ERC721_BRIDGE, "L721B-40");
+        internalRequire(address(_bridge.MESSENGER()) == address(_l1XDM), "L721B-50");
+        internalRequire(address(_bridge.messenger()) == address(_l1XDM), "L721B-60");
+        internalRequire(address(_bridge.superchainConfig()) == address(superchainConfig), "L721B-70");
+    }
+
+    function assertValidOptimismPortal(ISystemConfig _sysCfg) internal {
+        IOptimismPortal2 _portal = IOptimismPortal2(payable(_sysCfg.optimismPortal()));
+        internalRequire(stringEq(_portal.version(), optimismPortalVersion), "PORTAL-10");
+        internalRequire(bytecodeHashEq(address(_portal), optimismPortalHash), "PORTAL-20");
+
+        IDisputeGameFactory _dgf = IDisputeGameFactory(_sysCfg.disputeGameFactory());
+        internalRequire(address(_portal.disputeGameFactory()) == address(_dgf), "PORTAL-30");
+        internalRequire(address(_portal.systemConfig()) == address(_sysCfg), "PORTAL-40");
+        internalRequire(address(_portal.superchainConfig()) == address(superchainConfig), "PORTAL-50");
+        internalRequire(_portal.guardian() == superchainConfig.guardian(), "PORTAL-60");
+        internalRequire(_portal.paused() == superchainConfig.paused(), "PORTAL-70");
+        internalRequire(_portal.l2Sender() == defaultL2Sender, "PORTAL-80");
+    }
+
+    function assertValidDisputeGameFactory(ISystemConfig _sysCfg, address _l1PAO) internal {
+        IDisputeGameFactory _factory = IDisputeGameFactory(_sysCfg.disputeGameFactory());
+        internalRequire(stringEq(_factory.version(), disputeGameFactoryVersion), "DF-10");
+        internalRequire(_factory.owner() == address(_l1PAO), "DF-20");
+    }
+
+    function assertValidPermissionedDisputeGame(
```

Comment:
This should assert on the challenger role as well which currently must be the foundation multisig